### PR TITLE
Add support for MDL CSS Framework

### PR DESCRIPTION
**Pull request by drGrove**

### CODE REVIEW

File: test/template/test-bower.js [hunk 1]
```diff
@@ -157,6 +157,12 @@ describe('gulp-angular bower template', function () {
     result.should.not.match(/foundation/);
     result.should.not.match(/material/);
 
+    model.props.ui.key = 'mdl';
+    result = bower(model)
```
Comment (Kadajett) [inline test/template/test-bower.js]: Forgot a semicolon. :smile:

**Comment by zckrs** [comment]
Thank you.  :)

**Comment by zckrs** [comment]
Maybe we should propose option for https://github.com/jadjoubran/angular-material-design-lite ?

What do you think @Swiip @drGrove ?

**Comment by Swiip** [comment]
Wasn't so sure about adding a new ui framework but this one is great and your integration is also great. I can't honestly refuse this :) :+1:

**Comment by drGrove** [comment]
@zckrs,
MDL and Angular-Material are two completely different implementations. Unlike bootstrap and angular-ui-bootstrap (where the css is the same) MDL and Angular-Material are their own entities with very little cross over

**Comment by zckrs** [comment]
@drGrove Yeah MDL and Angular-Material are two different implementation

But I think about MDL and Angular-Material-Design-**Lite** 
Look https://github.com/jadjoubran/angular-material-design-lite/blob/master/bower.json#L28

**Comment by drGrove** [comment]
@zckrs,
AMDL is looking pretty good. May want to wait for a 1.0 release though.

**Comment by zckrs** [comment]
We should not wait integrate AMDL before merge this feature @Swiip ?

I am not sure one day AMDL release v1.0.0

Maybe @jadjoubran can help about this point.

**Comment by jadjoubran** [comment]
Hi,

Thanks for bringing this up!
AMDL is indeed a work in progress (0.3.0) at the time of writing.

Version 1.0 marks the integration of the most common components available in MDL + a fix for the following [issue #4](https://github.com/jadjoubran/angular-material-design-lite/issues/4) which requires an upstream fix.

But AMDL can be used today even though it hasn't reached v1.0.0 that's because if a component is not  available as a directive yet, you can just use the plain MDL component (same thing if you hit a limitation).

**Comment by drGrove** [comment]
@Swiip, 

What are your thoughts? Should I start working on an integration with AMDL as well?

**Comment by Swiip** [comment]
I think that even if AMDL is very young, it's interesting to offer a component library along with the choice of the UI framework so I think we should include it.

If you are willing to do the work, it's perfect, but just to be clear, you already did a great job and your PR will be merged even if you don't take this charge on you

**Comment by drGrove** [comment]
@Swiip,
Okay. I'll take a crack at the AMDL integration. Do you think that we should make Material Design the base key, then offer MDL, AMDL and Angular-Material as the sub-options (similar to bootstraps layout)

File: test/template/test-bower.js (hunk 1 above)
Comment (zckrs) [inline test/template/test-bower.js]: @drGrove  can you add semicolon ?

**Comment by zckrs** [comment]
@drGrove IMO the best solution is 
* MD: Angular-Material or Vanilla MD
* MDL: AMDL or Vanilla MDL

But at moment we can support only:
* Angular-Material
* AMDL

// @Swiip

**Comment by Swiip** [comment]
@drGrove Thank you very much for your work. Your work has been integrated in a new PR #743. Sorry if it's not your branch which is actually merged.

**Pull request closed without merging**